web/src/features/panels/search-panel/getSearchData.ts

```diff
@@ -29,6 +46,50 @@ export const getAllZones = (language: string) => {
   return zoneData;
 };
 
+// Hook to get all solar assets for search
+export const useGetSolarAssetsForSearch = () => {
+  const isSolarAssetsLayerEnabled = useAtomValue(isSolarAssetsLayerEnabledAtom);
+  const { data: geoJsonData, isLoading, error } = useGetSolarAssets();
+  const [solarAssets, setSolarAssets] = useState<SolarAssetRowType[]>([]);
+
+  useEffect(() => {
+    if (geoJsonData && 'features' in geoJsonData && Array.isArray(geoJsonData.features)) {
+      // Process GeoJSON features into a searchable format
+      const assets = geoJsonData.features
+        .filter(
+          (feature) =>
+            feature?.properties?.name &&
+            feature?.geometry?.type === 'Point' &&
+            Array.isArray(feature?.geometry?.coordinates) &&
+            isSolarAssetsLayerEnabled
+        )
+        .map((feature) => {
+          // Ensure we use the correct ID format that matches what the map expects
+          // The map layer is using 'name' as the promoteId in the source
+          const id = feature.properties.name;
+
+          return {
+            type: 'solar' as const,
+            id: String(id),
+            name: feature.properties.name,
+            country: feature.properties.country,
+            capacity: feature.properties.capacity_mw
+              ? `${Number.parseFloat(String(feature.properties.capacity_mw)).toFixed(
+                  1
+                )} MW`
+              : undefined,
+            coordinates: feature.geometry.coordinates.slice(0, 2) as [number, number],
+            status: feature.properties.status,
+          };
+        });
```

Comment:
These functions are going to be major performance bottle necks as it's more or less just two loops after each other and creating a massive object.

This should probably be done at build.